content/en/policies/best-practices/require-pod-requests-limits/require-pod-requests-limits.md

```diff
@@ -50,4 +50,18 @@ spec:
                 cpu: "?*"
               limits:
                 memory: "?*"
+          initContainers:
+          - resources:
+              requests:
+                memory: "?*"
+                cpu: "?*"
+              limits:
+                memory: "?*"
+          ephemeralContainers:
+          - resources:
+              requests:
+                memory: "?*"
+                cpu: "?*"
+              limits:
+                memory: "?*"
```

Comment:
This is not correct. These types of containers are optional and must be treated as conditional, else all submitted Pods will fail.

Reply:
Hi @chipzoller, 

Allow me some time to fix this.

Reply:
```
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: require-requests-limits
  annotations:
    policies.kyverno.io/title: Require Limits and Requests
    policies.kyverno.io/category: Best Practices, EKS Best Practices
    policies.kyverno.io/severity: medium
    policies.kyverno.io/subject: Pod
    policies.kyverno.io/minversion: 1.6.0
    policies.kyverno.io/description: >-
      As application workloads share cluster resources, it is important to limit resources
      requested and consumed by each Pod. It is recommended to require resource requests and
      limits per Pod, especially for memory and CPU. If a Namespace level request or limit is specified,
      defaults will automatically be applied to each Pod based on the LimitRange configuration.
      This policy validates that all containers have something specified for memory and CPU
      requests and memory limits.
spec:
  validationFailureAction: Audit
  background: true
  rules:
  - name: validate-container-resources
    match:
      any:
      - resources:
          kinds:
          - Pod
    validate:
      message: "CPU and memory resource requests and memory limits are required for containers."
      pattern:
        spec:
          containers:
          - resources:
              requests:
                memory: "?*"
                cpu: "?*"
              limits:
                memory: "?*"
          =(initContainers):
          - =(resources):
              requests:
                memory: "?*"
                cpu: "?*"
              limits:
                memory: "?*"
          =(ephemeralContainers):
          - =(resources):
              requests:
                memory: "?*"
                cpu: "?*"
              limits:
                memory: "?*"
```

Hi @chipzoller, I've made some updates to the policy. Could you please review it once? 
I've already checked this policy at Kyverno Playground, `initContainers` with resources & `initContainers` without resources.
cc: @realshuting

Reply:
You don't want the conditional anchor around `resources` because you're saying these are mandatory. But `initContainers` and `ephemeralContainers` are optional.

Reply:
```
    validate:
      message: "CPU and memory resource requests and memory limits are required for containers."
      pattern:
        spec:
          containers:
          - resources:
              requests:
                memory: "?*"
                cpu: "?*"
              limits:
                memory: "?*"
          =(initContainers):
          - resources:
              requests:
                memory: "?*"
                cpu: "?*"
              limits:
                memory: "?*"
          =(ephemeralContainers):
          - resources:
              requests:
                memory: "?*"
                cpu: "?*"
              limits:
                memory: "?*"
```

Got it, now? Looks Good?

Reply:
That looks correct.